### PR TITLE
Rewrite lever.rb in Lua

### PR DESCRIPTION
**Pull request by hawkowl**
I attempted to clean up the CLI to be a bit more consistent, too.

### CODE REVIEW

File: lever.lua [hunk 1]
```diff
@@ -0,0 +1,209 @@
+-- Inspect and pull levers
+world = df.global.world
```
Comment (myk002) [inline lever.lua:2]: what does this gain from being global?

File: lever.lua [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+function leverPullJob(lever, priority)
+    local ref = df.general_ref_building_holderst:new()
+    ref.building_id = lever.id
+
+    local job = df.job:new()
+    job.job_type = df.job_type.PullLever
+    job.pos = {
+        x = lever.centerx,
+        y = lever.centery,
+        z = z
```
Comment (myk002) [inline lever.lua:15]: I don't see where `z` is defined. should this be `lever.z`?

File: lever.lua [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    else
+        lever.state = 1
+    end
+
+    print(leverDescribe(lever))
+end
+
+function leverDescribe(lever)
+    local lever_name = ''
+    if string.len(lever.name) > 0 then
```
Comment (myk002) [inline lever.lua:46]:
```suggestion
    if #lever.name > 0 then
```

File: lever.lua [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+function ListLevers()
+    for k, v in ipairs(world.buildings.other.TRAP) do -- hint:df.building_trapst
+        if v.trap_type == df.trap_type.Lever then
+            print(('#%d: %s'):format(k, leverDescribe(v)))
+        end
+    end
+end
+
+function getLever(opts)
+    local lever = utils.binsearch(world.buildings.all, opts.pullId, 'id')
```
Comment (myk002) [inline lever.lua:116]:
```suggestion
    local lever = df.building.find(opts.pullId)
```

File: lever.lua [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+            return lever.trap_type
+        end) then
+            if not lever.trap_type == df.trap_type.Lever then
+                return nil
+            end
+        end
+        return lever
+    else
+        return nil
+    end
```
Comment (myk002) [inline lever.lua:129]:
```suggestion
    if df.building_trapst:is_instance(lever)
            and lever.trap_type == df.trap_type.Lever then
        return lever
    end
```

Comment (hawkowl) [reply lever.lua:129]: ooh much nicer

File: lever.lua (hunk 2 above)
Comment (hawkowl) [reply lever.lua:15]: ah whoops

File: lever.lua (hunk 1 above)
Comment (hawkowl) [reply lever.lua:2]: I cargoculted it from somewhere else, I think 🤔

File: docs/lever.rst [hunk 6]
```diff
@@ -9,27 +9,26 @@ Usage
...
+    Print out a list of your fort's levers, including their ID, name, activation
     state, and what they are linked to (if anything).
-``lever pull <id> [<options>]``
+``lever pull --id <id> [<options>]``
     Queue a job so a dwarf will pull the specified lever. This is the same as
     :kbd:`q` querying the building and queueing a :kbd:`P` pull job.
-
-If your levers aren't named, you can find out a lever's ID with the
-`bprobe <probe>` command.
+``lever show --id <id>``
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply docs/lever.rst:17]: I don't think we have the functionality to see what is selected in the UI, yet? If we do, good idea.

File: docs/lever.rst [hunk 7]
```diff
@@ -9,27 +9,26 @@ Usage
...
+``lever pull --id 42 --cheat``
     Skip the job and pull the lever with the hand of Armok!
 
 Options
 -------
 
 ``--priority``
     Queue a job at high priority.
-``--now``
+``--cheat``
```
[the remark this answers is not in the document]
Comment (hawkowl) [reply docs/lever.rst]: ack, switched out the wording to hopefully make it clearer that it's not just raising the job's priority

File: docs/lever.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (myk002) [reply docs/lever.rst:17]: soon, oh so soon: DFHack/dfhack#2577

File: docs/lever.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (myk002) [reply docs/lever.rst]: ah, ok. I understand why you changed it. that's good to clarify.

**Review by myk002 — APPROVED** [review]
Looks good! Would you like to make the ID optional for `pull` and `show`? The `dfhack.gui.getSelectedBuilding()` function will work by tomorrow

**Comment by myk002** [comment]
`dfhack.gui.getSelectedBuilding()` fixes have been merged

**Review by myk002 — APPROVED** [review]

**Pull request merged**